internal/ioutil/read_file.go

```diff
@@ -62,6 +62,23 @@ func ReadFileWithFileInfo(name string) ([]byte, fs.FileInfo, error) {
 //
 // passes NOATIME flag for reads on Unix systems to avoid atime updates.
 func ReadFile(name string) ([]byte, error) {
+	if !disk.ODirectPlatform {
+		// Don't wrap with un-needed buffer.
+		// Don't use os.ReadFile, since it doesn't pass NO_ATIME when present.
+		f, err := OpenFileDirectIO(name, readMode, 0o666)
```

Comment:
Shouldn't we use OpenFileDirectIO since disk.ODirectPlatform == false in this case ?

Reply:
Yeah. It falls back on the implementation, but you are completely correct.